### ISSUE TITLE
Add Minio To Aspire

### ISSUE THREAD

**Issue by josephaw1022** [dotnet/aspire#1801]
In order to fulfill that vision of being able to be completely cloud agnostic with Aspire, a cloud agnostic storage component is needed.

**Comment by davidfowl** [comment]
Can you motivate this some more? Why should this be a first class part of aspire?

**Comment by josephaw1022** [comment]
> Can you motivate this some more? Why should this be a first class part of aspire?

Minio is an open source s3 api compatible bucket storage solution that can be run on any cloud through containerization. 

Currently the only opinionated and first class solution is storage solution is Azure storage blob. However, some users using minio, aws s3, or s3gw will want a first class component that fits their need. 

So with minio, there is no need for a "s3 bucket" component because its a 1-1 of s3.

**Comment by davidfowl** [comment]
I think this proposal needs to be more fleshed out with a sample and an understanding of how deployment would work.

**Comment by josephaw1022** [comment]
@davidfowl I am figuring out the sdk side now 

And yes. And it should be supported because Aspire is a .net cloud native framework...

## Cloud Native Bucket Storage 
https://landscape.cncf.io/
![Screenshot (4)](https://github.com/dotnet/aspire/assets/47674962/237fa6cb-1c14-4350-93c0-80deb62f6355)

And per the docs of Aspire 

![image](https://github.com/dotnet/aspire/assets/47674962/c0f87aa3-5989-4140-a0bb-a06a282e0a5a)

**Comment by davidfowl** [comment]
Right, but we're not going to put all of the CNCF landscape in aspire. We're going to prioritize the "ubiquitous ones" and the community can fill the gaps for everything else. Then when something becomes more ubiquitous, we can consider it.

**Comment by josephaw1022** [comment]
@davidfowl I wasn't trying to bring CNCF up. I was just answering your question from earlier.

But AWS s3 itself has been around since 2006 and is the first cloud services ever offered by AWS and is incredibly popular. And Minio is the DeFacto oss version of s3. You can't get more ubiquitous than s3. 

And the minio sdk has no compatibility issues with AWS S3 itself. So, this would work with AWS users who want to target their s3 buckets.

**Comment by Reza-Noei** [comment]
Any update on this ?

@eerhardt

**Comment by rouke-broersma** [comment]
@davidfowl I think this issue is missing some context. AWS S3 and Minio are implementations of S3 compatible object storage. S3 is the ubiquitous interface for accessing object storage, supported at least for basic operations in most all cloud providers, even though AWS first created it for their S3 object storage offering. The only large cloud vendor not supporting S3 as object storage api format in any way is Azure. 

If this is ubiquitous enough to be considered is of course up for debate.

**Comment by davidfowl** [comment]
This is a great choice for the community toolkit https://github.com/CommunityToolkit/Aspire.

**Comment by glacasa** [comment]
I made a quick test to see if I could do it, so I thought I'd share : https://github.com/pericia/Pericia.Aspire.Minio

There is probably more work to do to make it clean, but it could be a nice starting point. Feel free to use it, until we get an official integration or one in the community toolkit

**Comment by davidfowl** [comment]
This one has both a hosting and client integration https://github.com/dotnet/aspire/pull/1800/

**Comment by eerhardt** [comment]
Aspire isn't taking new integrations but this is a great idea for the CommunityToolkit! https://github.com/CommunityToolkit/Aspire/issues

Or possibly could be part of https://github.com/aws/integrations-on-dotnet-aspire-for-aws/ since it is s3 api compatible.
